### PR TITLE
Remove image spec fork that was added to support ArgsEscaped non-OCI docker field

### PR DESCRIPTION
**Pull request by kiashok**
**Root Cause:**

Image-spec fork was added to our CRI branch to support the non-OCI docker image-spec field called 'ArgsEscaped'. ArgsEscaped is used by docker to determine how the image command should be interpreted. If ArgsEscaped is set, then the first element of the command array will be used as-is.
We would like to get rid of the dependency on the image-fork spec while we work on a cleaner fix that can be pushed upstream as well.

**Fix:**

Instead of relying on the image-fork spec and adding the ArgsEscaped field to our image spec, we try to read the ArgsEscaped value by making an extra json.Unmarshal call during getImage()

**Testing**

Manually tested the changes to make sure that the fix does not regress any behavior that the original fix was made for. I will be adding regression tests to the upcoming PR that has the complete upstream fix.

### CODE REVIEW

File: pkg/server/container_create_windows.go [hunk 1]
```diff
@@ -299,7 +300,8 @@ func removeMount(s *runtimespec.Spec, dest string) {
 }
 
 func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxPid uint32, netnsPath string, config *runtime.ContainerConfig,
-	sandboxConfig *runtime.PodSandboxConfig, sandboxPlatform string, image *imagespec.Image) (*runtimespec.Spec, error) {
+	sandboxConfig *runtime.PodSandboxConfig, sandboxPlatform string, Image *imagestore.Image) (*runtimespec.Spec, error) {
```
Comment (kevpar) [inline pkg/server/container_create_windows.go]: We shouldn't use upper-case variable name in Go for function arguments. Can we rename to something else?

File: pkg/store/image/image.go [hunk 2]
```diff
@@ -144,12 +148,25 @@ func getImage(ctx context.Context, i containerd.Image) (*Image, error) {
 		return nil, errors.Wrapf(err, "unmarshal image config %s", rb)
 	}
 
+	type imageExtended struct {
```
Comment (kevpar) [inline pkg/store/image/image.go:155]: It would be helpful to have a comment here explaining why we need to do this.

**Comment by kevpar** [comment]
> Since this is only a temporary fix that will very soon be replaced by the cleaner upstream fix, I will be adding regression tests to the upcoming PR instead of this one.

Does this refer to the fact that once https://github.com/opencontainers/image-spec/pull/892 gets in, we can remove this change?

**Comment by kiashok** [comment]
> > Since this is only a temporary fix that will very soon be replaced by the cleaner upstream fix, I will be adding regression tests to the upcoming PR instead of this one.
> 
> Does this refer to the fact that once [opencontainers/image-spec#892](https://github.com/opencontainers/image-spec/pull/892) gets in, we can remove this change?

I was referring more to your WIP changes that I am looking into as your changes seem to handle all cases and I was wondering if we just wanted to have the upstream fix in our fork too and that's why I mentioned it. But after giving this a second thought, it should not be a problem to leave these changes as is since we longer have dependency on image-spec fork.

**Comment by kevpar** [comment]
> > > Since this is only a temporary fix that will very soon be replaced by the cleaner upstream fix, I will be adding regression tests to the upcoming PR instead of this one.
> > 
> > 
> > Does this refer to the fact that once [opencontainers/image-spec#892](https://github.com/opencontainers/image-spec/pull/892) gets in, we can remove this change?
> 
> I was referring more to your WIP changes that I am looking into as your changes seem to handle all cases and I was wondering if we just wanted to have the upstream fix in our fork too and that's why I mentioned it. But after giving this a second thought, it should not be a problem to leave these changes as is since we longer have dependency on image-spec fork.

I think even with my WIP changes, we still need a way to determine if ArgsEscaped is set on the image config, so we will still need this change (or ArgsEscaped added to image-spec). IIRC this same change was part of my WIP.

**Comment by kiashok** [comment]
> > > > Since this is only a temporary fix that will very soon be replaced by the cleaner upstream fix, I will be adding regression tests to the upcoming PR instead of this one.
> > > 
> > > 
> > > Does this refer to the fact that once [opencontainers/image-spec#892](https://github.com/opencontainers/image-spec/pull/892) gets in, we can remove this change?
> > 
> > 
> > I was referring more to your WIP changes that I am looking into as your changes seem to handle all cases and I was wondering if we just wanted to have the upstream fix in our fork too and that's why I mentioned it. But after giving this a second thought, it should not be a problem to leave these changes as is since we longer have dependency on image-spec fork.
> 
> I think even with my WIP changes, we still need a way to determine if ArgsEscaped is set on the image config, so we will still need this change (or ArgsEscaped added to image-spec). IIRC this same change was part of my WIP.

I updated the description. Since I am in the process of adding regression tests to your WIP code changes, I'll check them in as part of that PR.

File: pkg/server/container_create.go [hunk 3]
```diff
@@ -42,7 +42,8 @@ func init() {
 
 // setOCIProcessArgs sets process args. It returns error if the final arg list
 // is empty.
-func setOCIProcessArgs(g *generator, config *runtime.ContainerConfig, image *imagespec.Image) error {
+func setOCIProcessArgs(g *generator, config *runtime.ContainerConfig, imagestoreImage *imagestore.Image) error {
```
Comment (kevpar) [inline pkg/server/container_create.go]: Maybe a name like `criImage` would be more natural here? `imagestoreImage` sounds a bit awkward.

File: pkg/server/container_create_windows.go [hunk 4]
```diff
@@ -299,7 +300,8 @@ func removeMount(s *runtimespec.Spec, dest string) {
 }
 
 func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxPid uint32, netnsPath string, config *runtime.ContainerConfig,
-	sandboxConfig *runtime.PodSandboxConfig, sandboxPlatform string, image *imagespec.Image) (*runtimespec.Spec, error) {
+	sandboxConfig *runtime.PodSandboxConfig, sandboxPlatform string, imagestoreImage *imagestore.Image) (*runtimespec.Spec, error) {
```
Comment (kevpar) [inline pkg/server/container_create_windows.go]: Same change here to `criImage`

File: vendor.conf [hunk 5]
```diff
@@ -37,7 +37,7 @@ github.com/moby/sys 74ec3fe1325a016151d353bb6654d960042f9e96
 github.com/modern-go/concurrent 1.0.3
 github.com/modern-go/reflect2 1.0.1
 github.com/opencontainers/go-digest c9281466c8b2f606084ac71339773efd177436e7
-github.com/opencontainers/image-spec deb02d24daef68ac4a88d8f9883ba03b75bdc187 https://github.com/kevpar/image-spec.git # fork
+github.com/opencontainers/image-spec v1.0.1
```
Comment (kevpar) [inline vendor.conf:40]: I think this will need to update `go.mod` as well now that we've moved away from vndr. Is that right @katiewasnothere?

Comment (kevpar) [reply vendor.conf:40]: Or maybe this also needs to be rebased on the new `windows_port`.

**Review by helsaawy — COMMENTED** [review]
If I am understanding correctly, the blob data in the content store is set by docker, so we don't need to set that when pulling an image, right?

**Comment by kiashok** [comment]
> If I am understanding correctly, the blob data in the content store is set by docker, so we don't need to set that when pulling an image, right?

That is right. We are only trying to read argsEscaped from the blob data we got in

**Review by helsaawy — APPROVED** [review]

**Review by kevpar — APPROVED** [review]
LGTM

**Pull request merged**